src/test/ceph_crypto.cc

```diff
@@ -269,6 +269,11 @@ void do_simple_crypto() {
 }
 
 #if GTEST_HAS_DEATH_TEST && !defined(_WIN32)
+
+#ifndef __has_feature
```

Comment:
@xxhdx1985126 according to https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005ffeature.html , 

> Note that __has_feature and __has_extension are not recommended for use in new code, and are only provided for compatibility with Clang

i'd sugget reference https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005ffeature.html and https://clang.llvm.org/docs/LanguageExtensions.html#has-feature-and-has-extension in the commit message.

Reply:
Done, please take a look again:-)